modules/compression/src/bundle.js

```diff
@@ -1,7 +1,34 @@
-/* global window, global */
-const moduleExports = require('./index');
-const _global = typeof window === 'undefined' ? global : window;
-// @ts-ignore
-_global.loaders = _global.loaders || {};
-// @ts-ignore
-module.exports = Object.assign(_global.loaders, moduleExports);
+import * as moduleExports from './index';
+
+import {createWorker} from '@loaders.gl/worker-utils';
+import ZlibDeflateTransform from './lib/zlib/zlib-deflate-transform';
+import ZlibInflateTransform from './lib/zlib/zlib-inflate-transform';
+
+import LZ4DeflateTransform from './lib/zlib/zlib-deflate-transform';
+import LZ4InflateTransform from './lib/zlib/zlib-inflate-transform';
+
+// Enable the bundle to be imported directly, exporting all module symbols
+export * from './index';
+
+// Support for import as an HTML script. Add to global `loaders` object
+globalThis.loaders = globalThis.loaders || {};
+Object.assign(globalThis.loaders, moduleExports);
+
+// Create worker (only starts when script is loaded in a worker)
+// Note: Since all module code is already part of the bundle, the overhead is minimal
```

Comment:
Is this true? I see that `createWorker` always defines `self.onmessage`, which will affect the main thread as well.

Reply:
Need to get a test setup for the scripts, but both `createWorker` and `createLoaderWorker` start with 

```typescript
  // Check that we are actually in a worker thread
  if (typeof self === 'undefined') {
    return;
  }
```

Reply:
`self` is available on the main thread as well.

Reply:
Good catch - changed to `isWorker` and added a test case for that value on main thread.